### PR TITLE
doc/cephadm: Add troubleshooting guidance for ssh connection failures. [cephadm]

### PR DESCRIPTION
**Pull request by zdover23**
For more thorough information about this commit, see:
https://tracker.ceph.com/issues/44905

Signed-off-by: Zac Dover <zac.dover@gmail.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: doc/cephadm/troubleshooting.rst [hunk 1]
```diff
@@ -102,3 +102,41 @@ Manually running containers
...
+     [root@mon1 ~]# cat key | cephadm shell -- ceph cephadm set-ssk-key -i -
+
+2. Ensure that the ssh config is correct::
+   
+     [root@mon1 ~]# cephadm shell -- ceph cephadm get-ssh-config > config
+
+3. Verify that we can connect to the host::
+    
+     [root@mon1 ~]# ssh -F config -i key root@mon1i
+     ssh: connect to host mon1 port 22: Connection timed out
```
Comment (sebastian-philipp) [inline doc/cephadm/troubleshooting.rst:140]: In this case, `ssh: connect to host mon1 port 22: Connection timed out` was an error message by a user. Hopefully, users now get a root shell on mon1.

Comment (zdover23) [reply doc/cephadm/troubleshooting.rst]: I've removed this line.

File: doc/cephadm/troubleshooting.rst [hunk 2]
```diff
@@ -102,3 +102,40 @@ Manually running containers
...
+
+Error message::
+
+  xxxxxx.gateway_bootstrap.HostNotFound: -F /tmp/cephadm-conf-kbqvkrkw root@10.10.1.2
+  raise OrchestratorError('Failed to connect to %s (%s).  Check that the host is reachable and accepts  connections using the cephadm SSH key' % (host, addr)) from
+  orchestrator._interface.OrchestratorError: Failed to connect to 10.10.1.2 (10.10.1.2).  Check that the host is reachable and accepts connections using the cephadm SSH key
+
+Things users can do:
+
+1. Ensure that it is possible to connect to the host via ssh::
```
Comment (sebastian-philipp) [inline doc/cephadm/troubleshooting.rst:118]:
```suggestion
1. Ensure cephadm has an SSH identity key::
```

File: doc/cephadm/troubleshooting.rst [hunk 3]
```diff
@@ -102,3 +102,40 @@ Manually running containers
...
+
+2. Ensure that the ssh config is correct::
+   
+     [root@mon1 ~]# cephadm shell -- ceph cephadm get-ssh-config > config
+
+3. Verify that we can connect to the host::
+    
+     [root@mon1 ~]# ssh -F config -i key root@mon1
+
+4. There is a limitation right now (as of April 2020): the ssh user is root. This is hardcoded.
```
Comment (sebastian-philipp) [inline doc/cephadm/troubleshooting.rst:141]:
```suggestion
4. There is a limitation right now: the ssh user is always `root`.
```

**Pull request merged**